src/runtime/Codecs/FunctionCodec.cs

```diff
@@ -0,0 +1,377 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Python.Runtime.Codecs
+{
+    //like MethodWrapper but not static, so we can throw some state into it.
```

Comment:
Would like to see this combined with the MethodWrapper in methodwrapper.cs if possible.  In my case I want to wrap an member function rather than a static function.  @amos402 @filmor @lostmsu